node/src/components/linear_chain.rs

```diff
@@ -444,58 +451,30 @@ where
                 effect_builder
-                    .is_bonded_validator(fs.era_id, fs.public_key)
-                    .map(|is_bonded| {
-                        if is_bonded {
-                            Ok((maybe_signatures, fs, is_bonded))
-                        } else {
-                            Err((maybe_signatures, fs))
-                        }
-                    })
-            }
-            .result(
-                |(maybe_signatures, fs, is_bonded)| {
-                    Event::IsBonded(maybe_signatures, fs, is_bonded)
-                },
-                |(maybe_signatures, fs)| Event::IsBondedFutureEra(maybe_signatures, fs),
-            ),
-            Event::IsBondedFutureEra(maybe_signatures, fs) => {
-                match self.latest_block.as_ref() {
-                    // If we don't have any block yet, we cannot determine who is bonded or not.
-                    None => effect_builder
-                        .immediately()
-                        .event(move |_| Event::IsBonded(maybe_signatures, fs, false)),
-                    Some(block) => {
-                        let latest_header = block.header();
-                        let state_root_hash = latest_header.state_root_hash();
-                        // TODO: Use protocol version that is valid for the block's height.
-                        let protocol_version = ProtocolVersion::new(SemVer::V1_0_0);
-                        effect_builder
-                            .is_bonded_in_future_era(
-                                *state_root_hash,
-                                fs.era_id,
-                                protocol_version,
-                                fs.public_key,
-                            )
-                            .map(|res| {
-                                match res {
-                                    // Promote this error to a non-error case.
-                                    // It's not an error that we can't find the era that this
-                                    // FinalitySignature is for.
-                                    Err(error) if error.is_era_validators_missing() => Ok(false),
-                                    _ => res,
-                                }
-                            })
-                            .result(
-                                |is_bonded| Event::IsBonded(maybe_signatures, fs, is_bonded),
-                                |error| {
-                                    error!(%error, "is_bonded_in_future_era returned an error.");
-                                    panic!("couldn't check if validator is bonded")
-                                },
-                            )
-                    }
-                }
+                    .is_bonded_validator(
+                        fs.public_key,
+                        fs.era_id,
+                        activation_era_id,
+                        initial_state_root_hash,
```

Comment:
Is it possible to get `activation_era_id`, `initial_state_root_hash` from the Chainspec loader?